[Ng Yu Soon] iP

YuSoonZ opened the pull request:
#TheAdvisor
> "To have a great idea, have a lot of them." - Thomas Edison [source](https://www.brainyquote.com/quotes/thomas_a_edison_149040)

TheAdvisor frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

YuSoonZ committed:
Changed the name of the chatbot + added a new intro + outro to the bot

YuSoonZ committed:
Added an echo function

YuSoonZ committed:
Added the function to give a list of strings which will be output when user inputs "list"
Removed the echo function

YuSoonZ committed:
Added a new helper class: Task to store the task description and related functions
TheAdvisor class now can mark/unmark the list that has been input

YuSoonZ committed:
Add support for tracking three types of tasks:

1. ToDos: tasks without any date/time attached to it e.g., visit new theme park
2. Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
3. Events: tasks that start at a specific date/time and ends at a specific date/time
e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

YuSoonZ committed:
Added Exception handling for wrong input/format

YuSoonZ committed:
Added a delete function to remove item from the list

YuSoonZ committed:
Updated with enum feature instead of if-else statement

YuSoonZ committed:
Tested, and passed all inputs

YuSoonZ committed:
Added a save function to store list from previous iteration

YuSoonZ committed:
Added a date + time function for event and deadline tasks

YuSoonZ committed:
Added Ui class to handle interaction with user

YuSoonZ committed:
Added run() method to TheAdvisor class

YuSoonZ committed:
Added Storage implementation into TheAdvisor class to save/load data

YuSoonZ committed:
Added Parse functionality to parse user inputs
Removed Prompts

YuSoonZ committed:
Added TaskList class that handles tasklist related commands

YuSoonZ committed:
Added the classes into TheAdvisor package

YuSoonZ committed:
Added gradle support

YuSoonZ committed:
Added JUnit Tests

YuSoonZ committed:
Packaged the App as a JAR file

YuSoonZ committed:
Added Javadocs

YuSoonZ committed:
Updated code to comply with coding standards

YuSoonZ committed:
Add find function

YuSoonZ committed:
Add find function

YuSoonZ committed:
Revert "Add find function"

This reverts commit bff0de800d5f2a8068a000f357e03851c8f649a5.

YuSoonZ committed:
Merge commit '2b3d30b2eedaf256f983f2ec51fd3671c01e259a'

sethteo commented:
Detailed comments for all methods given, followed convention for naming as well as imports, correct usage of braces. :))

YuSoonZ committed:
Implement CheckStyle for code quality checks

sethteo reviewed (commented):
Look at the most updated review below :)

YuSoonZ committed:
Add JavaFX GUI for TheAdvisor

YuSoonZ committed:
Add varargs

YuSoonZ committed:
Update list method in Ui class

sethteo commented on src/main/java/TheAdvisor/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class Deadline extends Task implements Serializable {
+
+    // protected String by;
```
Perhaps you could remove this

sethteo commented on src/main/java/TheAdvisor/Task.java:
```diff
@@ -0,0 +1,76 @@
+     * @return "X" if the task is done, " " if it is not done.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
Maybe add the comment to the javadoc rather than a inline comment

sethteo reviewed (commented):
LGTM! Code follows coding standard where variables are in camelCase and classes are in PascalCase. Good use of comments which explain the methods well. Ordering of imports across files are consistent. Keep it up :)

Tsenrae commented on src/main/java/TheAdvisor/TheAdvisor.java:
```diff
@@ -0,0 +1,164 @@
+     * @return The response generated by the application.
+     * @throws TheAdvisorException If there is an error processing the user input.
+     */
+    public String getResponse(String req) throws TheAdvisorException {
```
It's maybe better to have the code to parse the command in the Parser class

Tsenrae reviewed (commented):
LGTM! Overall, code is quite clean and easy to read. Good usage of OOP abstraction

YuSoonZ committed:
Generate new JAR file

The JAR file is generated using the 'shadowJar' task in Gradle, ensuring that all dependencies are bundled into a single executable JAR.

YuSoonZ committed:
Add assert statements to document critical assumptions

Utilize the assert feature, not JUnit assertions, to document important assumptions throughout the codebase. These assertions serve to clarify expected conditions and validate critical assumptions at various points in the code.

YuSoonZ committed:
Update list.bin

YuSoonZ committed:
Enhance code quality and general code improvements
- Updated code to have better code quality: Naming wise and conventions
- Updated methods to give cleaner outputs to UI

YuSoonZ committed:
Add stream implementation for operation in Tasklist class and for task list printing in Ui class

Description:
Utilise stream to improve functionality when printing list and when finding items from a list

YuSoonZ committed:
Merge pull request #1 from YuSoonZ/heads/A-Assertions

Add assert statements to document critical assumptions

YuSoonZ committed:
Merge tag 'A-Streams'

YuSoonZ committed:
Update find method in task list to use stream

YuSoonZ committed:
Resolved merge conflicts

YuSoonZ committed:
Update code to remove merge conflict

YuSoonZ committed:
Increase flexbility in search for items even if it matches keyword partially

YuSoonZ committed:
Update code to comply with SLAP standard

YuSoonZ committed:
Update DialogBox class for better GUI interface

YuSoonZ committed:
Add personality to the chatbot

YuSoonZ committed:
Add implementation to handle repeated tasks

YuSoonZ committed:
Add additional class to do more automated test

YuSoonZ committed:
Update README.md

YuSoonZ committed:
Add userguide

MarcusGitty commented on src/main/java/TheAdvisor/Main.java:
```diff
@@ -0,0 +1,31 @@
+ */
+public class Main extends Application {
+
+    private TheAdvisor advisor = new TheAdvisor();
```
I like how you named the class "TheAdvisor" it is very clear

MarcusGitty commented on src/main/java/TheAdvisor/MainWindow.java:
```diff
@@ -0,0 +1,60 @@
+     * the dialog container. Clears the user input after processing.
+     */
+    @FXML
+    private void handleUserInput() throws TheAdvisorException {
```
Good naming of the method !

MarcusGitty commented on src/main/java/TheAdvisor/TaskList.java:
```diff
@@ -0,0 +1,172 @@
+     * @param index The index to be checked.
+     * @return True if the index is within bounds and the task list is not empty, false otherwise.
+     */
+    public boolean checkIndex(int index) {
```
Would indexChecker(int index) be a better name ?

MarcusGitty reviewed (commented):
All the best for your IP,  good use of comments,  Jiayous !!

YuSoonZ committed:
Update readme

YuSoonZ committed:
Update readme to fix readability

YuSoonZ committed:
Fix readme bug

YuSoonZ committed:
Fix readme bug
